fms/datasets/__init__.py

```diff
@@ -23,3 +23,98 @@ def get_dataset(name: str, tokenizer: BaseTokenizer, data: str = "", **kwargs):
         raise NameError(f"Dataset name should be one of {__dataset_factory.keys()}")
     ctr = __dataset_factory[name]
     return ctr(data, tokenizer, **kwargs)
+
+
+def _state_dict_save_helper(target):
+    if isinstance(target, dict):
+        dict_attrs = target
+    else:
+        dict_attrs = target.__dict__
```

Comment:
I'm assuming here that a Dataset always has `__dict__` implemented the way this is expecting it?

Reply:
nah it's that every python object has a `__dict__`, so this is a catch-all